subsys/bluetooth/controller/ll_sw/lll_scan.h

```diff
@@ -31,6 +31,9 @@ struct lll_scan {
 	uint16_t duration_expire;
 	uint8_t  phy:3;
 	uint8_t  is_adv_ind:1;
+
+	/* XXX: used with lll scheduling */
+	struct lll_scan_aux *lll_aux;
```

Comment:
Correct me, this is only needed as initiator to remember the CONNECT_REQ, instead move the `node_conn_rx` into `struct lll_scan` ?

Reply:
This is needed when aux is scheduled from LLL.
For example, if aux was scheduled from LLL on ext PDU, once that ext PDU reaches ULL we will allocate lll_scan_aux as usual. However, since we skip ULL scheduling and thus do not pass that information to LLL, we need to store it somewhere and then retrieve when scanned aux reaches ULL. Similar process is when we scheduled aux to aux from LLL, so 1st aux stores lll_scan_aux and 2nd aux retrieves it.